File: sdk/go/dotenv.go
package sdk

import (
	"bufio"
	"os"
	"strings"
)

// LoadDotEnv loads KEY=VALUE lines (plus # comments, optional quotes) from
// the given .env files. Existing environment variables always win.
func LoadDotEnv(paths ...string) {
	for _, path := range paths {
		if path == "" {
			continue
		}
		f, err := os.Open(path)
		if err != nil {
			continue
		}
		sc := bufio.NewScanner(f)
		for sc.Scan() {
			line := strings.TrimSpace(sc.Text())
			if line == "" || strings.HasPrefix(line, "#") {
				continue
			}
			eq := strings.Index(line, "=")
			if eq <= 0 {
				continue
			}
			key := strings.TrimSpace(line[:eq])
			value := strings.TrimSpace(line[eq+1:])
			value = strings.Trim(value, `"`)
			if key != "" && os.Getenv(key) == "" {
				os.Setenv(key, value)
			}
		}
		f.Close()
	}
}

File: sdk/ts/src/dotenv.ts
// .env loading — mirror of sdk/dotenv.nim and sdk/go/dotenv.go.
// KEY=VALUE lines (plus # comments, optional quotes); existing shell env
// always wins.

import * as fs from "fs";

export function loadDotEnv(...paths: string[]): void {
  for (const path of paths) {
    if (!path) continue;
    let text: string;
    try {
      text = fs.readFileSync(path, "utf8");
    } catch {
      continue;
    }
    for (const rawLine of text.split("\n")) {
      const line = rawLine.trim();
      if (line === "" || line.startsWith("#")) continue;
      const eq = line.indexOf("=");
      if (eq <= 0) continue;
      const key = line.slice(0, eq).trim();
      let value = line.slice(eq + 1).trim();
      if (value.startsWith('"') && value.endsWith('"')) {
        value = value.slice(1, -1);
      }
      if (key !== "" && process.env[key] === undefined) {
        process.env[key] = value;
      }
    }
  }
}
